### PR TITLE
LPS-163783 Extract Content Dashboard Item API to the content-dashboard-api module

### PR DESCRIPTION
**Pull request by cgoncas**
Motivation
=======
What is the problem this work solves, including
[LPS-161020 Add the Preview Action to the Version History for Web Content](https://issues.liferay.com/browse/LPS-161020)
[LPS-163783 Extract Content Dashboard Item API to the content-dashboard-api module](https://issues.liferay.com/browse/LPS-163783)

Proposed Solution
========
I have extracted the interfaces ContentDashboardItem, ContentDashboardItemFactory, ContentDashboardItemActionFactoryTracker and ContentDashboardItemSubtypeFactoryTracker to content-dashboard-api module.

All the Blogs,D&M and Journal code have been moved to their own modules, so now content-dashboard-wed dones't have any dependency with the assets implementation

Steps to Verify:
----------------
No functional changes, all test must be green.

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:tango

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:tango

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

File: modules/apps/content-dashboard/content-dashboard-web/build.gradle [hunk 1]
```diff
@@ -21,15 +21,13 @@ dependencies {
 	compileOnly project(":apps:asset:asset-categories-api")
 	compileOnly project(":apps:asset:asset-display-page-api")
```
Comment (jkappler) [inline modules/apps/content-dashboard/content-dashboard-web/build.gradle:22]: @cgoncas, was testing, and we can remove  compileOnly project(":apps:asset:asset-display-page-api") and ompileOnly project(":apps:layout:layout-display-page-api") too

**Comment by jkappler** [comment]
@cgoncas, we are missing com.liferay.content.dashboard.item export in bnd in content-dashboard-api

**Comment by jkappler** [comment]
other than that, all good ❤️

**Comment by cgoncas** [comment]
> @cgoncas, we are missing com.liferay.content.dashboard.item export in bnd in content-dashboard-api

Fixed :), thanks!!!

**Comment by cgoncas** [comment]
Hey @jkappler !!

I have rebase this PR with https://github.com/liferay-tango/liferay-portal/pull/2862 and there are not any conflicts, so I am going to forward it today.

Thanks!!

**Comment by cgoncas** [comment]
ci:reevaluate:1317401_5142

**Comment by jkappler** [comment]
ci:test:tango

**Comment by jkappler** [comment]
ci:test:sf

**Comment by jkappler** [comment]
@cgoncas, sending on top of versions

**Comment by jkappler** [comment]
@cgoncas, update process for AssetVocabulary fails, most likely because we need to add a dependency to the model. Will take a look at it

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:tango

**Comment by cgoncas** [comment]
ci:test:relevant

**Pull request closed without merging**